tsdb/executor.go

```diff
@@ -625,23 +625,23 @@ func (e *SelectExecutor) processFunctions(results [][]interface{}, columnNames [
 }
 
 func (e *SelectExecutor) processSelectors(results [][]interface{}, callPosition int, hasTimeField bool, columnNames []string) ([][]interface{}, error) {
-	for i, vals := range results {
-		for j := 1; j < len(vals); j++ {
-			switch v := vals[j].(type) {
+	// if the columns doesn't have enough columns, expand it
+	for i, columns := range results {
+		if len(columns) != len(columnNames) {
```

Comment:
I'll be honest, I have no clue how this fixes the issue. :-)

Perhaps you can explain.